libnetwork/firewall_linux_test.go

```diff
@@ -64,25 +57,32 @@ func TestUserChain(t *testing.T) {
 			assert.NilError(t, err)
 			defer c.Stop()
 
-			// init. condition, FORWARD chain empty DOCKER-USER not exist
-			assert.Check(t, is.DeepEqual(getRules(t, iptable4, fwdChainName), []string{"-P FORWARD ACCEPT"}))
-			assert.Check(t, is.DeepEqual(getRules(t, iptable6, fwdChainName), []string{"-P FORWARD ACCEPT"}))
+			// init. condition
+			golden.Assert(t, getRules(t, iptable4, fwdChainName),
+				fmt.Sprintf("TestUserChain_iptables-%v_append-%v_fwdinit4", tc.iptables, tc.append))
+			golden.Assert(t, getRules(t, iptable4, fwdChainName),
+				fmt.Sprintf("TestUserChain_iptables-%v_append-%v_fwdinit6", tc.iptables, tc.append))
 
-			if tc.insert {
-				_, err = iptable4.Raw("-A", fwdChainName, "-j", "DROP")
+			if tc.append {
+				_, err := iptable4.Raw("-A", fwdChainName, "-j", "DROP")
 				assert.Check(t, err)
 				_, err = iptable6.Raw("-A", fwdChainName, "-j", "DROP")
 				assert.Check(t, err)
 			}
 			arrangeUserFilterRule()
 
-			assert.Check(t, is.DeepEqual(getRules(t, iptable4, fwdChainName), tc.fwdChain))
-			assert.Check(t, is.DeepEqual(getRules(t, iptable6, fwdChainName), tc.fwdChain))
-			if tc.userChain != nil {
-				assert.Check(t, is.DeepEqual(getRules(t, iptable4, usrChainName), tc.userChain))
-				assert.Check(t, is.DeepEqual(getRules(t, iptable6, usrChainName), tc.userChain))
+			golden.Assert(t, getRules(t, iptable4, fwdChainName),
+				fmt.Sprintf("TestUserChain_iptables-%v_append-%v_fwdafter4", tc.iptables, tc.append))
+			golden.Assert(t, getRules(t, iptable6, fwdChainName),
+				fmt.Sprintf("TestUserChain_iptables-%v_append-%v_fwdafter6", tc.iptables, tc.append))
+
+			if tc.iptables {
+				golden.Assert(t, getRules(t, iptable4, fwdChainName),
+					fmt.Sprintf("TestUserChain_iptables-%v_append-%v_fwdafter4", tc.iptables, tc.append))
+				golden.Assert(t, getRules(t, iptable6, fwdChainName),
+					fmt.Sprintf("TestUserChain_iptables-%v_append-%v_fwdafter6", tc.iptables, tc.append))
```

Comment:
The original code was checking `userChainName` (ie. `DOCKER-USER`), but here you're checking the same chain as above (ie. `FORWARD`).